ec2-cluster/main.tf

```diff
@@ -28,15 +27,14 @@ module "nodes" {
   bucket_name          = var.bucket_name
 }
 
-module "cluster-provisioner" {
+module "cluster_provisioner" {
   source            = "./modules/cluster-provisioner"
-  provisioner_count = var.instance_count
   instance_ips      = module.nodes.instance_ips
```

Comment:
What we can do more there is to zip `instance_ips` with `instance_keys` and provide a single variable so inside `cluster_provisioner` we are sure that both have the same length.

Reply:
I think that `instance_ips` isn't a variable, but an output.

Reply:
Yes, `instance_ips` is the output related to `count` param which is set to `length(instance_keys)` so everything is dependent.

Reply:
Maybe we should actually make output with already zipped ips with keys 🤔